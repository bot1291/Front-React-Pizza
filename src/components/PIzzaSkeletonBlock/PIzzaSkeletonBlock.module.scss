@import '../../styles/variables';

.pizzaBlock {
	width: 280px;
	text-align: center;
}

.image {
	width: 260px;
}

.title {
	width: 70%;
	height: 24px;
	margin: 0 auto;
	margin-bottom: 20px;
	background-color: $skeleton-main-color;
	border-radius: 10px;
}

.selector {
	display: flex;
	flex-direction: column;
	height: 85px;
	padding: 6px;
	background-color: $skeleton-main-color;
	border-radius: 10px;
}

.info {
	display: flex;
	justify-content: space-between;
	align-items: center;
	margin-top: 20px;
}

.price {
	width: 30%;
	height: 30px;
	background-color: $skeleton-main-color;
	border-radius: 10px;
}

.list {
	display: flex;
	flex: 1;

	&:first-of-type {
		margin-bottom: 6px;
	}

	div {
		flex: 1;
		padding: 8px;

		&.active {
			background: $skeleton-white;
			border-radius: 5px;
			box-shadow: 0 2px 4px rgba(0, 0, 0, 0.04%);
		}
	}
}

.plus {
	width: 30px;
	height: 30px;
	background-color: $skeleton-main-color;
	border-radius: 50%;

	&:hover {
		fill: $black;
	}
}

.buttonsBlock {
	display: flex;

	> div + div {
		margin-left: 10px;
	}
}

.button {
	display: flex;
	justify-content: space-between;
	min-width: 95px;
	height: 50px;
	padding: 9px;
	border: 1px solid transparent;
	border-radius: 30px;

	&,
	span {
		color: $skeleton-white;
	}

	&.buttonOutline {
		background-color: $skeleton-white;
		border-width: 5px;
		border-color: $skeleton-main-color;
	}
}

.buttonTitle {
	width: 65%;
	height: 100%;
	background-color: $skeleton-main-color;
	border-radius: 10px;
}

.count {
	width: 30%;
	height: 100%;
	background-color: $skeleton-main-color;
	border-radius: 50%;
}

@media screen and (max-width: 450px) {
	.pizzaBlock {
		width: 100%;
	}

	.image {
		width: 100%;
	}
}
