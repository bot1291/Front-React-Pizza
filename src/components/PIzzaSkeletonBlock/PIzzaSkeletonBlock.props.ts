import { DetailedHTMLProps, HTMLAttributes } from 'react';

export interface PIzzaSkeletonBlockProps
	extends DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement> {}
