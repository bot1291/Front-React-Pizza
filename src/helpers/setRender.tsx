import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createReduxStore } from '../store/store';

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
const setRender = <T,>(Component: JSX.Element, store?: Record<string, T>) => {
	return render(
		<Provider store={createReduxStore(store)}>{Component}</Provider>
	);
};

export default setRender;
