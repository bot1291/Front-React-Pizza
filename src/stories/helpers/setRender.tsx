import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createReduxStore } from '../../store/store';

const setRender = <T,>(
	Component: JSX.Element,
	store?: Record<string, T>
): JSX.Element => {
	return (
		<MemoryRouter>
			<Provider store={createReduxStore(store)}>{Component}</Provider>
		</MemoryRouter>
	);
};

export default setRender;
