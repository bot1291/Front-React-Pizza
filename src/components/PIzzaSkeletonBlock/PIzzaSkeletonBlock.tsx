import styles from './PIzzaSkeletonBlock.module.scss';
import { PIzzaSkeletonBlockProps } from './PIzzaSkeletonBlock.props';
import cn from 'classnames';
import { FC } from 'react';

export const PIzzaSkeletonBlock: FC<PIzzaSkeletonBlockProps> = ({
	className,
	...props
}) => {
	return (
		<div
			data-testid="wrapper"
			className={cn(className, styles.pizzaBlock)}
			{...props}>
			<img
				className={cn(styles.image)}
				src="https://i.imgur.com/mTJhAis.png"
				alt={'Loading pizza'}
			/>
			<div className={cn(styles.title)} />
			<div className={cn(styles.selector)}>
				<div className={cn(styles.list)}>
					<div className={cn(styles.active)} />
					<div />
				</div>
				<div className={cn(styles.list)}>
					<div className={cn(styles.active)} />
					<div />
				</div>
			</div>
			<div className={cn(styles.info)}>
				<div className={cn(styles.price)} />
				<div className={styles.buttonsBlock}>
					<div className={cn(styles.plus)}></div>
					<div className={cn(styles.plus)}></div>
				</div>
				<div className={cn(styles.button, styles.buttonOutline)}>
					<div className={styles.buttonTitle} />
					<div className={styles.count} />
				</div>
			</div>
		</div>
	);
};
