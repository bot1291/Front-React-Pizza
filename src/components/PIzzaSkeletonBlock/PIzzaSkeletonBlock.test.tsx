import setRender from '../../helpers/setRender';
import { PIzzaSkeletonBlock } from './PIzzaSkeletonBlock';
import { screen } from '@testing-library/dom';

describe('PIzzaSkeletonBlock', () => {
	it('Should follow the structuce', () => {
		setRender(<PIzzaSkeletonBlock />);
		const wrapperElem = screen.getByTestId('wrapper');
		expect(wrapperElem).toMatchSnapshot();
	});
});
