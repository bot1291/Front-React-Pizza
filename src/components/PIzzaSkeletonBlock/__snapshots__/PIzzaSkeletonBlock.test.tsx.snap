// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PIzzaSkeletonBlock Should follow the structuce 1`] = `
<div
  class="pizzaBlock"
  data-testid="wrapper"
>
  <img
    alt="Skeleton"
    class="image"
    src="https://i.imgur.com/mTJhAis.png"
  />
  <div
    class="title"
  />
  <div
    class="selector"
  >
    <div
      class="list"
    >
      <div
        class="active"
      />
      <div />
    </div>
    <div
      class="list"
    >
      <div
        class="active"
      />
      <div />
    </div>
  </div>
  <div
    class="info"
  >
    <div
      class="price"
    />
    <div
      class="buttonsBlock"
    >
      <div
        class="plus"
      />
      <div
        class="plus"
      />
    </div>
    <div
      class="button buttonOutline"
    >
      <div
        class="buttonTitle"
      />
      <div
        class="count"
      />
    </div>
  </div>
</div>
`;
